screens/Scan/ScanScreen.tsx

```diff
@@ -26,6 +26,10 @@ export const ScanScreen: React.FC = () => {
     })();
   }, [isBluetoothOn]);
 
+  useEffect(() => {
+    if (!controller.isEmpty) controller.START_PERMISSION_CHECK();
+  });
```

Comment:
Use deps array to run use effect only when controller.isEmpty has any change else this will run on every render